### PR TITLE
synchronize bpf watch map reads

### LINKED ISSUE #3505: Data race in bpf module
```
teleport[29239]: fatal error: concurrent map read and map write
teleport[29239]: goroutine 381 [running]:
teleport[29239]: runtime.throw(0x2019d08, 0x21)
teleport[29239]: #011/opt/go/src/runtime/panic.go:774 +0x72 fp=0xc000974d98 sp=0xc000974d68 pc=0x43cf12
teleport[29239]: runtime.mapaccess2_fast64(0x1cbf5c0, 0xc00087aea0, 0x10000000e, 0x1bb4a60, 0xc00060d180)
teleport[29239]: #011/opt/go/src/runtime/map_fast64.go:61 +0x1b5 fp=0xc000974dc0 sp=0xc000974d98 pc=0x41f3a5
teleport[29239]: github.com/gravitational/teleport/lib/bpf.(*Service).emitDiskEvent(0xc00026cc00, 0xc00067a280, 0x12c, 0x12c)
teleport[29239]: #011/gopath/src/github.com/gravitational/teleport/lib/bpf/bpf.go:323 +0x117 fp=0xc000974e40 sp=0xc000974dc0 pc=0xf140f7
teleport[29239]: github.com/gravitational/teleport/lib/bpf.(*Service).loop(0xc00026cc00)
teleport[29239]: #011/gopath/src/github.com/gravitational/teleport/lib/bpf/bpf.go:227 +0x2b7 fp=0xc000974fd8 sp=0xc000974e40 pc=0xf13257
teleport[29239]: runtime.goexit()
teleport[29239]: #011/opt/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc000974fe0 sp=0xc000974fd8 pc=0x46cfe1
teleport[29239]: created by github.com/gravitational/teleport/lib/bpf.New
teleport[29239]: #011/gopath/src/github.com/gravitational/teleport/lib/bpf/bpf.go:148 +0x606
teleport[29239]: goroutine 1 [select]:
teleport[29239]: github.com/gravitational/teleport/lib/service.(*TeleportProcess).WaitForSignals(0xc00021c1e0, 0x2385940, 0xc000052070, 0x0, 0x0)
teleport[29239]: #011/gopath/src/github.com/gravitational/teleport/lib/service/signals.go:77 +0x3f2
(...)
systemd[1]: teleport.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
systemd[1]: teleport.service: Failed with result 'exit-code'.
```

### PR DESCRIPTION
**Pull request by fspmarshall**
Fixes #3505

### CODE REVIEW

**Review by russjones — APPROVED** [review]

**Comment by fspmarshall** [comment]
retest this please

**Comment by fspmarshall** [comment]
retest this please

**Review by klizhentas — APPROVED** [review]

File: lib/bpf/bpf.go [hunk 1]
```diff
@@ -457,6 +457,13 @@ func (s *Service) emit6NetworkEvent(eventBytes []byte) {
 	ctx.AuditLog.EmitAuditEvent(events.SessionNetwork, eventFields)
 }
 
+func (s *Service) getWatch(cgoupID uint64) (ctx *SessionContext, ok bool) {
```
Comment (webvictim) [inline lib/bpf/bpf.go:460]: `cgroupID`

File: lib/bpf/bpf.go [hunk 2]
```diff
@@ -457,6 +457,13 @@ func (s *Service) emit6NetworkEvent(eventBytes []byte) {
...
+	s.watchMu.Lock()
+	defer s.watchMu.Unlock()
+	ctx, ok = s.watch[cgoupID]
```
Comment (webvictim) [inline lib/bpf/bpf.go:463]: `cgroupID`

**Review by webvictim — APPROVED** [review]

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Pull request merged**